### ISSUE TITLE
Corrupted Python installation should not be used by plugin

### ISSUE THREAD

**Issue by alanhamlett** [wakatime/jetbrains-wakatime#120]
Error from `idea.log` file:

```
2019-04-16 10:31:25,193 [entQueue-0]   INFO -                       WakaTime - Initializing WakaTime plugin v9.3.2 (https://wakatime.com/)
2019-04-16 10:31:25,197 [entQueue-0]  DEBUG -                       WakaTime - Logging level set to DEBUG
2019-04-16 10:31:25,325 [entQueue-0]  DEBUG -                       WakaTime - com.sun.jna.platform.win32.Win32Exception: The system cannot find the file specified.
2019-04-16 10:31:25,330 [entQueue-0]  DEBUG -                       WakaTime - com.sun.jna.platform.win32.Win32Exception: The system cannot find the file specified.
2019-04-16 10:31:25,334 [entQueue-0]  DEBUG -                       WakaTime - pythonw --version
2019-04-16 10:31:25,540 [entQueue-0]  DEBUG -                       WakaTime - Python 3.7.0
2019-04-16 10:31:25,540 [entQueue-0]  DEBUG -                       WakaTime - Found python binary: pythonw
2019-04-16 10:31:34,142 [d thread 1]  DEBUG -                       WakaTime - wakatime cli version check output: "Fatal Python error: initfsencoding: unable to load the file system codec  File "C:\Python27\Lib\encodings\__init__.py", line 123    raise CodecRegistryError,\                            ^SyntaxError: invalid syntaxCurrent thread 0x00005728 (most recent call first):"
2019-04-16 10:31:34,142 [d thread 1]  DEBUG -                       WakaTime - wakatime cli version check exit code: -1073740791
2019-04-16 10:31:34,142 [d thread 1]   INFO -                       WakaTime - Upgrading wakatime-cli ...
2019-04-16 10:31:47,761 [d thread 1]   INFO -                       WakaTime - Finished upgrading wakatime-cli.
2019-04-16 10:31:47,761 [d thread 1]  DEBUG -                       WakaTime - CLI location: C:\Users\PCUSER\AppData\Roaming\WakaTime\wakatime-master\wakatime\cli.py
2019-04-16 10:33:07,327 [entQueue-0]   INFO -                       WakaTime - Finished initializing WakaTime plugin
2019-04-16 10:33:25,543 [2-thread-1]  DEBUG -                       WakaTime - Executing CLI: [pythonw, C:\Users\PCUSER\AppData\Roaming\WakaTime\wakatime-master\wakatime\cli.py, --entity, C:/Users/PCUSER/AndroidStudioProjects/project1/app/src/main/java/com/example/Main.java, --time, 1555400001.6330, --language, JAVA, --plugin, AndroidStudio/3.3.2 AndroidStudio-wakatime/9.3.2]
2019-04-16 10:33:28,373 [2-thread-1]  DEBUG -                       WakaTime - Fatal Python error: initfsencoding: unable to load the file system codec
2019-04-16 10:33:28,373 [2-thread-1]  DEBUG -                       WakaTime -   File "C:\Python27\Lib\encodings\__init__.py", line 123
2019-04-16 10:33:28,373 [2-thread-1]  DEBUG -                       WakaTime -     raise CodecRegistryError,\
2019-04-16 10:33:28,373 [2-thread-1]  DEBUG -                       WakaTime -                             ^
2019-04-16 10:33:28,373 [2-thread-1]  DEBUG -                       WakaTime - SyntaxError: invalid syntax
2019-04-16 10:33:28,373 [2-thread-1]  DEBUG -                       WakaTime -
2019-04-16 10:33:28,373 [2-thread-1]  DEBUG -                       WakaTime - Current thread 0x00000ae4 (most recent call first):
2019-04-16 10:33:28,373 [2-thread-1]  DEBUG -                       WakaTime - Command finished with return value: -1073740791
```

The plugin shouldn't be considered initialized until the version of wakatime-cli can be read. In this case, the `pythonw wakatime-cli --version` command failed so the WakaTime plugin assumed wakatime-cli was missing and re-downloaded it.

**Comment by alanhamlett** [comment]
Something strange is `Python 3.7.0` is used, but the error comes from `C:\Python27\Lib\encodings\__init__.py`. This could mean a corrupted Python installation, but in any case the plugin should detect that and avoid using this installation of Python. At least for windows we already download embedded Python if no system version is found, which would fix this.

**Comment by tariqywsf** [comment]
Is there a way  u can remove or clean the failed or corrupted python installation to see if it fixes the wakatime plugin and work correctly?

**Comment by alanhamlett** [comment]
Yes, you could uninstall either Python2 or Python3 or both if they're not needed by anything on the machine. Then the WakaTime plugin would use it's embedded Python.

**Comment by tariqywsf** [comment]
@alanhamlett i uninstalled Python2 and i can see wakatime reports on my dashboard again!
but when i check the plugin status i still see there is new version available and when i check the plugin in Android Studio i can't see any update avaialable yet(my wakatime version is 9.3.2 ). but the plugin is working again thats the good news!

**Comment by alanhamlett** [comment]
> when i check the plugin status i still see there is new version available

That's probably from old heartbeats syncing, since they would send the original wakatime-cli version.

**Comment by alanhamlett** [comment]
Fixed in v11.0.0 using standalone wakatime-cli that doesn't require system Python to be installed nor working.
